src/ray/raylet/test/local_object_manager_test.cc

```diff
@@ -0,0 +1,242 @@
+                  }
+                }) {}
+
+  size_t free_objects_batch_size = 3;
+  std::shared_ptr<MockWorkerClient> owner_client;
+  rpc::CoreWorkerClientPool client_pool;
+  MockIOWorkerPool worker_pool;
+  MockObjectInfoAccessor object_table;
+  LocalObjectManager manager;
+
+  std::unordered_set<ObjectID> freed;
+};
+
+TEST_F(LocalObjectManagerTest, TestPin) {
+  rpc::Address owner_address;
+  owner_address.set_worker_id(WorkerID::FromRandom().Binary());
+
+  std::vector<ObjectID> object_ids;
+  std::vector<std::unique_ptr<RayObject>> objects;
+
+  for (size_t i = 0; i < free_objects_batch_size; i++) {
+    ObjectID object_id = ObjectID::FromRandom();
+    object_ids.push_back(object_id);
+    std::string meta = std::to_string(static_cast<int>(rpc::ErrorType::OBJECT_IN_PLASMA));
+    auto metadata = const_cast<uint8_t *>(reinterpret_cast<const uint8_t *>(meta.data()));
+    auto meta_buffer = std::make_shared<LocalMemoryBuffer>(metadata, meta.size());
+    std::unique_ptr<RayObject> object(
+        new RayObject(nullptr, meta_buffer, std::vector<ObjectID>()));
+    objects.push_back(std::move(object));
+  }
+  manager.PinObjects(owner_address, object_ids, std::move(objects));
+
+  for (size_t i = 0; i < free_objects_batch_size; i++) {
+    ASSERT_TRUE(freed.empty());
+    ASSERT_TRUE(owner_client->ReplyObjectEviction());
+  }
+  std::unordered_set<ObjectID> expected(object_ids.begin(), object_ids.end());
+  ASSERT_EQ(freed, expected);
+}
+
+TEST_F(LocalObjectManagerTest, TestExplicitSpill) {
+  rpc::Address owner_address;
+  owner_address.set_worker_id(WorkerID::FromRandom().Binary());
+
+  std::vector<ObjectID> object_ids;
+  std::vector<std::unique_ptr<RayObject>> objects;
+
+  for (size_t i = 0; i < free_objects_batch_size; i++) {
+    ObjectID object_id = ObjectID::FromRandom();
+    object_ids.push_back(object_id);
+    std::string meta = std::to_string(static_cast<int>(rpc::ErrorType::OBJECT_IN_PLASMA));
+    auto metadata = const_cast<uint8_t *>(reinterpret_cast<const uint8_t *>(meta.data()));
+    auto meta_buffer = std::make_shared<LocalMemoryBuffer>(metadata, meta.size());
+    std::unique_ptr<RayObject> object(
+        new RayObject(nullptr, meta_buffer, std::vector<ObjectID>()));
+    objects.push_back(std::move(object));
+  }
+  manager.PinObjects(owner_address, object_ids, std::move(objects));
+
+  int num_times_fired = 0;
+  manager.SpillObjects(object_ids, [&](const Status &status) mutable {
+    ASSERT_TRUE(status.ok());
+    num_times_fired++;
+  });
+  ASSERT_EQ(num_times_fired, 0);
+
+  EXPECT_CALL(worker_pool, PushIOWorker(_));
+  std::vector<std::string> urls;
+  for (size_t i = 0; i < object_ids.size(); i++) {
+    urls.push_back("url" + std::to_string(i));
+  }
+  ASSERT_TRUE(worker_pool.io_worker_client->ReplySpillObjects(urls));
+  ASSERT_EQ(num_times_fired, 1);
+  for (size_t i = 0; i < object_ids.size(); i++) {
+    ASSERT_EQ(object_table.object_urls[object_ids[i]], urls[i]);
+  }
```

Comment:
It looks like `AsyncRestoreSpilledObject ` is not tested here. Are you planning to add it in the follow-up PR?

Reply:
Oh I didn't add it because there isn't that much to test...but yeah let me add a unit test for it.